pkg/services/ngalert/schedule/compat.go

```diff
@@ -38,12 +31,15 @@ func FromAlertStateToPostableAlerts(logger log.Logger, firingStates []*state.Sta
 			nA["__value_string__"] = alertState.Results[0].EvaluationString
 		}
 
-		genURL := appURL
-		if uid := nL[ngModels.RuleUIDLabel]; len(uid) > 0 && u != nil {
-			oldPath := u.Path
+		var urlStr string
+		if uid := nL[ngModels.RuleUIDLabel]; len(uid) > 0 && appURL != nil {
+			u := *appURL
```

Comment:
`net.URL` contains pointers so this won't copy of all data, but since `Path` is a string this is OK.

Reply:
correct, the only pointer is UserInfo which is immutable, according to the documentation